packages/@sanity/base/src/components/rovingFocus/useRovingFocus.ts

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import {useCallback, useEffect, useState} from 'react'
+import {RovingFocusProps} from '.'
+
+const EMPTY_ARRAY = []
+
+const MUTATION_ATTRIBUTE_FILTER = ['aria-hidden', 'disabled', 'href']
+
+const FOCUSABLE =
+  'a[href], button, input, textarea, select, details, [tabindex]:not([tabindex="-1"])'
+
+function getFocusableElements(element: HTMLElement) {
+  return [...(element.querySelectorAll(FOCUSABLE) as any)].filter(
+    (el) => !el.hasAttribute('disabled') && !el.getAttribute('aria-hidden="true"')
+  ) as HTMLElement[]
+}
+
+export function useRovingFocus(props: RovingFocusProps, dependencies?: unknown[]): undefined {
+  const {direction = 'horizontal', initialFocus, loop = true, pause = false, rootElement} = props
+  const [focusedIndex, setFocusedIndex] = useState<number>(-1)
+  const [focusableElements, setFocusableElements] = useState<HTMLElement[]>([])
+
+  const deps = dependencies ? dependencies : EMPTY_ARRAY
```

Comment:
🍬 i think you could provide a default param instead of the ternary

Reply:
Yes, definitely, will do 👍